components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -5324,14 +5324,15 @@ def __init__(self, config, prefix, webIDLFile):
 
         # Do codegen for all the typdefs
         for t in typedefs:
-            if t.innerType.isUnion():
-                cgthings.extend([CGGeneric("\npub type %s = %s;\n\n" % (t.identifier.name,
-                                                                        "UnionTypes::" + str(t.innerType)))])
-            else:
-                assert not typeNeedsRooting(t.innerType, config.getDescriptorProvider)
-                cgthings.extend([CGGeneric("\npub type %s = " % (t.identifier.name)),
-                                 getRetvalDeclarationForType(t.innerType, config.getDescriptorProvider()),
-                                 CGGeneric(";\n\n")])
+            for e in enums:
+                if t.innerType.isUnion():
+                    cgthings.extend([CGGeneric("\npub use self::%s as %s;\n\n" % (e.identifier.name,
+                                    (t.identifier.name)))])
+                else:
+                    assert not typeNeedsRooting(t.innerType, config.getDescriptorProvider)
+                    cgthings.extend([CGGeneric("\npub type %s = " % (t.identifier.name)),
```

Comment:
Even if it's not a Union typedef, I think we should still re-export it, instead of using the `pub type = blah` syntax.